codeforces_api/api_request_maker.py

```diff
@@ -5,34 +5,39 @@ class CodeforcesApiRequestMaker():
 
     api_key = ""
     secret = ""
-    randomKey = 0
+    rand = 0
     anonimus = False
 
-    def __init__(self, api_key=None, secret=None, rand=1000000):
+    def __init__(self, api_key=None, secret=None, rando=1000000):
         """
-        Initializes main variables: api_key, secret, random (default is between 1 and 1000000),
-        but when you set it it will be yours.
+        Initializes main variables: api_key, secret, random (default is between 100000
+        and 1000000 unless specified)
         """
 
-        if rand == 1000000:
-            rand = random.randint(1, 1000000)
+        if rando >= 1000000 or rando <= 99999:
+            rando = random.randint(100000, 999999)
         if api_key is None and secret is None:
             self.anonimus = True
         else:
             self.api_key = api_key
             self.secret = secret
             self.anonimus = False
-        self.randomKey = rand
+        self.rand = rando
 
     def generate_url(self, method_name, **fields):
         """
         Generates request URL for API.
         """
+
         if not self.anonimus:
+
+            #Renew Rand
+            self.renew_rand()
```

Comment:
<img width="881" alt="image" src="https://user-images.githubusercontent.com/3719664/76145607-e49e5300-609b-11ea-96aa-ba7181c4a690.png">

https://codeforces.com/apiHelp

Based on the guidelines for apiSig:
1. "First six characters of the apiSig parameter can be arbitrary." I thought this meant **only** six characters, especially given the example. I have however commented out the line which guarantees only six digits, and replaced with 1-6 digits for now.
2. "We recommend to choose them at random for each request. Let's denote them as rand" previously a **rand** was generated only once for each _CodeforcesApi_ instance and it never changed. So I decided to update it for each new request. While I did try to add in arguments for allowing a custom **rand**, I felt that it would mess up the existing functions and just made it always random.

With that in mind, what do you suggest we do?

Reply:
I think, that if random was said we need to use it always. And should be at least 6 digits. If it's less than 6 digits it will be error returned that signature is not valid.

Reply:
Hmmm... I'll see what I can do without breaking _too much_

Reply:
Just create a variable in a class, which indicates if random_number was set or not

Reply:
Just checked for authentication to confirm, indeed it expects 6 and only 6 digits. I think I fixed everything in the last commit but you should double-check